test/dlc_tests/ec2/test_smdebug.py

```diff
@@ -18,13 +18,21 @@
 @pytest.mark.parametrize("ec2_instance_type", SMDEBUG_EC2_GPU_INSTANCE_TYPE, indirect=True)
 @pytest.mark.flaky(reruns=0)
 def test_smdebug_gpu(training, ec2_connection, region, ec2_instance_type, gpu_only, py3_only):
+    smdebug_test_timeout = 2400
+    if is_tf_version("1", training):
+        if is_nightly_context():
+            smdebug_test_timeout = 7200
+        else:
+            pytest.skip("TF1 gpu smdebug tests can take up to 2 hours, thus we are only running in nightly context")
```

Comment:
nit: Is there a way to make this a `pytest.mark.skip` decorator rather than a `skip` that is triggered after the test has already started? If it can be converted to a skip decorator, then the EC2 instance for the test would never be started.

Reply:
good point, there are several ways we could do this but it's not clear enough without over-scoping this change and adding more resistance to @Teng-xu's change. Let's handle this in a future PR.